src/FormSerializationWriter.cs

```diff
@@ -206,24 +226,25 @@ public void WriteCollectionOfPrimitiveValues<T>(string? key, IEnumerable<T>? val
     public void WriteCollectionOfEnumValues<T>(string? key, IEnumerable<T?>? values) where T : struct, Enum
 #endif
     {
-        if(values == null) return;
+        if (values is null) return;
 
-        StringBuilder? valueNames = null;
-        foreach(var x in values)
+        using (var enumerator = values.GetEnumerator())
         {
-            if(x.HasValue && Enum.GetName(typeof(T), x.Value) is string valueName)
+            if (!enumerator.MoveNext()) return;
+        }
+
+        List<string> nonNullValues = new List<string>();
+        foreach (var value in values)
+        {
+            if (value.HasValue)
```

Comment:
I'm not sure why we changed that condition? This will break enums serialization where the symbol name is different from the wirename